### PR TITLE
perf: No longer using the styler package to style the code

### PR DESCRIPTION
**Pull request by krlmlr**
With only a few tweaks I get down to 9 test failures, pushed to this PR. Some of them look interesting.

To finish this:

- ~~Revert the last commit~~
- Fix remaining tests (lazytest can help)
- Remove styler entirely

For #49.

### CODE REVIEW

**Comment by krlmlr** [comment]
@moodymudskipper: I get inconsistent results between my local install and CI/CD. I'm running R 4.2, could this be a reason?

Would you like to finish this off?

**Comment by moodymudskipper** [comment]
package versions can also be the problem since I have tests that construct gepolt objects for instance, they should be run conditionally if given package version is installed.

Re indention I see in my snapshot test failures : 

* wrong indention after function call `fun(` going to the line
* When a string goes to the next line it should not be indented (e.g. read.table constructor, and function defs we get from secrets), technically should also be the case for names between backquotes or infix ops that go to the line but unlikely to happen. Even if we ignore the issues of nested quotes in backquotes `%` we still have to deal with `'`, `"` and their escaped counterparts so we basically need to build a poor man's parser I think, go character by character, flag when we open a string, and find where we close it. this needs to be reapplied at every step, i.e. we can't just have indent = paste. Or maybe we can repair at last step only, by subtracting the amount of incorrect indents once.
* current implementation has inconsistent space before native pipe
* indention not always applied
 
I can take it, but not in the short term

**Comment by krlmlr** [comment]
The weird formatting issues are a styler problem:

``` r
requireNamespace("styler")
structure(
  c("(function(x) {\n  x\n}) |>", "  (`environment<-`)(.GlobalEnv)"),
  class = "vertical"
)
#> (function(x) {
#>   x
#> }) |>
#>   x`environment<-`)(.GlobalEnv)
```

<sup>Created on 2023-07-07 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by krlmlr** [comment]
This is now looking much better, only indentation issues left. I can start using this for my purposes, we want to fix these issues at some point.

**Comment by krlmlr** [comment]
This builds on top of #204 and is now good enough for me. We need to think what to do about the final remaining indentation errors.

**Comment by krlmlr** [comment]
The first two commits now *should* be applicable on top of the main branch.

File: tests/testthat/_snaps/s3-ggplot2-ggplot.md [hunk 1]
```diff
@@ -44,16 +44,16 @@
...
-              collapse_labels_lines(labels)
-            }
-          }) |>
+        labels <- lapply(labels, as.character)
+        if (multi_line) {
+          labels
+        } else {
+          collapse_labels_lines(labels)
+        }
+      }) |>
```
Comment (moodymudskipper) [inline tests/testthat/_snaps/s3-ggplot2-ggplot.md:53]: same as above

File: tests/testthat/_snaps/s3-ggplot2-ggplot.md [hunk 2]
```diff
@@ -44,16 +44,16 @@
...
             (`environment<-`)(asNamespace("ggplot2")) |>
             structure(class = c("function", "labeller"))
         ) +
         ggplot2::xlim(1, 7) +
-        ggplot2::ylim(10, 45)
+          ggplot2::ylim(10, 45)
```
Comment (moodymudskipper) [inline tests/testthat/_snaps/s3-ggplot2-ggplot.md:58]: can be solved in .cstr_construct.ScalesList

**Review by moodymudskipper — CHANGES_REQUESTED** [review]
I identified the issues and will fix, will undo the ordered modification since it seems to me to be out of scope

**Comment by moodymudskipper** [comment]
snapshots should be ok now. now phasing out styler

**Comment by moodymudskipper** [comment]
@krlmlr This is is ready to merge as far as I'm concerned, do you want to review ?
* deparse_call gets a new collapse arg
* we add the class "constructive_code" which is basically "vertical" without styler
* styler and "vertical" are not used anymore
* a couple tweaks
* prettycode is moved to imports since it's not implied by styler anymore 

After this I propose a `maybe_highlight()` function to highlight only if prettycode is installed, an we can move it back to Suggests

**Review by moodymudskipper — APPROVED** [review]

**Comment by moodymudskipper** [comment]
one snapshot works differently for ubuntu-22.04 because of an indention issue, not sure what the root issue is

**Comment by krlmlr** [comment]
Thanks. How is the `"constructive_code"` class different from the `"glue"` class offered by {glue}? Seems good otherwise.

File: R/s3-ggplot2-Scale.R [hunk 3]
```diff
@@ -62,6 +62,8 @@
       args$palette <- "identity"
     } else if (identical(args$palette, quote(abs_area(max_size)))) {
       args$palette <- sprintf("scales::abs_area(%s)", environment(as.list(x)$palette)$max)
+    } else if (identical(args$palette, quote(rescale_pal(range)))) {
+      args$palette <- "scales::rescale_pal(range)"
```
Comment (moodymudskipper) [inline R/s3-ggplot2-Scale.R]: range is undefined so that won't work, I pushed a fix

**Review by moodymudskipper — APPROVED** [review]
This should work with the fix I just pushed, we didn't spot the actual ident issue though so we might see some issues arise with the next R version. These will be easier to investigate in time.

**Comment by moodymudskipper** [comment]
The glue class doesn't highlight code, "constructive_code" uses {prettycode} and is really just the "vertical" class, except no need for styler.

**Comment by krlmlr** [comment]
Thanks! Would you like to push to CRAN?

**Pull request merged**